corehq/ex-submodules/couchforms/models.py

```diff
@@ -338,6 +339,8 @@ def top_level_tags(self):
         return to_return
 
     def archive(self, user=None):
+        if self.is_archived:
+            raise BadOperationException("This form is already archived")
```

Comment:
What's the value of raising an exception here rather than performing a no-op like `return None`?  Is your point that if we ever try to archive a form that's already archived, it should be considered a bug?